assets/src/web-stories-block/block/block-types/selected-stories/storyPicker.js

```diff
@@ -0,0 +1,340 @@
+
+const StoryPickerModal = styled(Modal)`
+  width: calc(100% - 16px - 16px);
+  height: calc(100% - 60px - 60px);
+  position: relative;
+  overflow: hidden;
+`;
+
+const ModalContent = styled.div`
+  position: relative;
+  height: calc(100% - 93px);
+  margin: -24px -24px 0;
+  padding: 0;
+`;
+
+const ModalContentInner = styled.div`
+  padding: 12px 24px 24px;
+`;
+
+const ModalFooter = styled.div`
+  position: fixed;
+  bottom: 0;
+  z-index: 2;
+  width: 100%;
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  background: #f3f3f3;
+  border-top: 1px solid #ddd;
+  margin: 0 -24px;
+  padding: 10px 24px;
+
+  .components-button {
+    margin-left: 8px;
+  }
+`;
+
+const ModalFooterLeft = styled.div`
+  min-width: 1px;
+`;
+
+const LimitIndicator = styled.span(
+  ({ reachedMaximum }) => `
+  font-weight: 600;
+  color: ${reachedMaximum ? '#EB5757' : 'inherit'};
+`
+);
+
+const ModalFooterRight = styled.div``;
+
+function StoryPicker({
+  selectedStories,
+  setSelectedStories,
+  selectedStoriesObject,
+  setSelectedStoriesObject,
+  closeStoryPicker,
+  isSortingStories,
+  setIsSortingStories,
+}) {
+  const [fetchingForTheFirstTime, setFetchingForTheFirstTime] = useState(true);
+  const [currentAuthor, setCurrentAuthor] = useState([]);
+  const [authorKeyword, setAuthorKeyword] = useState('');
+  const { maxNumOfStories } = useConfig();
+
+  const {
+    fetchStories,
+    stories,
+    storiesOrderById,
+    totalPages,
+    allPagesFetched,
+    isLoading,
+    fetchAuthors,
+    authorSuggestions,
+  } = useApi(
+    ({
+      actions: {
+        storyApi: { fetchStories },
+        usersApi: { fetchAuthors },
+      },
+      state: {
+        stories: {
+          stories,
+          storiesOrderById,
+          totalPages,
+          allPagesFetched,
+          isLoading,
+        },
+        authorSuggestions,
+      },
+    }) => ({
+      fetchStories,
+      stories,
+      storiesOrderById,
+      totalPages,
+      allPagesFetched,
+      isLoading,
+      fetchAuthors,
+      authorSuggestions,
+    })
+  );
+
+  const { filter, page, search, sort, view } = useStoryView({
+    filters: STORY_STATUSES,
+    totalPages,
+  });
+
+  const fetchWebStories = useCallback(async () => {
+    const query = {
+      page: page.value,
+      searchTerm: search.keyword,
+      sortDirection: view.style === VIEW_STYLE.LIST && sort.direction,
+      sortOption: sort.value,
+    };
+
+    if (currentAuthor.length) {
+      query.author = parseInt(currentAuthor.pop().id);
+    }
+
+    await fetchStories(query);
+    setFetchingForTheFirstTime(false);
+  }, [
+    page.value,
+    search.keyword,
+    view.style,
+    sort,
+    currentAuthor,
+    fetchStories,
+    setFetchingForTheFirstTime,
+  ]);
+
+  const fetchStoryAuthors = useCallback(async () => {
+    const query = {
+      searchTerm: authorKeyword,
+    };
+
+    await fetchAuthors(query);
+  }, [authorKeyword, fetchAuthors]);
```

Comment:
Do you need to pass `fetchAuthors` in here?

Reply:
it is expecting us to provide this as a dependency, else the eslint flags as `react-hooks/exhaustive-deps`.

Reply:
Are you sure? A function doesn't change in the same way of variable does.

Reply:
I had not added it as dep, thinking the same but eslint complains about the above-mentioned rule.

Reply:
It's a function coming from another hook, not another module, so the linter rightfully complains.

All good here :-)